src/Cryptol/TypeCheck/Solver/Numeric.hs

```diff
@@ -137,6 +140,17 @@ tryGeqThanK _ t (Nat k) =
   -- XXX: K1 ^^ n >= K2
 
 
+-- (x >= 2 && x^a >= x^b) => a >= b
+tryGeqExp :: Ctxt -> Type -> Type -> Match Solved
+tryGeqExp _ x y = 
+      do  (x_1, a) <- (|^|) x
+          (x_2, b) <- (|^|) y
+          guard (x_1 == x_2)
+          n <- aNat x_1
```

Comment:
Small thing, but it might be better to move the `aNat` check earlier, perhaps right after the binding for `x_1`.  This should avoid the rest of the stuff (e.g., comparing the types) if we don't have a number in the base.

Reply:
Updated in [110a753](https://github.com/GaloisInc/cryptol/pull/1713/commits/110a753a94506829ac11402ccf4f0067bb18bebe)